src/Compression/CompressionCodecZSTD.cpp

```diff
@@ -79,9 +93,31 @@ void registerCodecZSTD(CompressionCodecFactory & factory)
 
             level = literal->value.safeGet<UInt64>();
             if (level > ZSTD_maxCLevel())
-                throw Exception("ZSTD codec can't have level more that " + toString(ZSTD_maxCLevel()) + ", given " + toString(level), ErrorCodes::ILLEGAL_CODEC_PARAMETER);
+                throw Exception(
+                    "ZSTD codec can't have level more that " + toString(ZSTD_maxCLevel()) + ", given " + toString(level),
+                    ErrorCodes::ILLEGAL_CODEC_PARAMETER);
+            if (arguments->children.size() > 1)
+            {
+                const auto * window_literal = children[1]->as<ASTLiteral>();
+                if (!window_literal)
+                    throw Exception("ZSTD codec second argument must be integer", ErrorCodes::ILLEGAL_CODEC_PARAMETER);
+
+                const int window_log = window_literal->value.safeGet<UInt64>();
+
+                ZSTD_bounds window_log_bounds = ZSTD_cParam_getBounds(ZSTD_c_windowLog);
+                if (ZSTD_isError(window_log_bounds.error))
+                    throw Exception(
+                        "ZSTD windowLog paramater is not supported " + std::string(ZSTD_getErrorName(window_log_bounds.error)),
+                        ErrorCodes::ILLEGAL_CODEC_PARAMETER);
+                if (window_log != 0 && (window_log > window_log_bounds.upperBound || window_log < window_log_bounds.lowerBound))
```

Comment:
Why should the `0` be special value for boundary check?

Reply:
value 0 means 'use default' in libzstd, i will add comment